Dangerfile

```diff
@@ -0,0 +1,150 @@
+# frozen_string_literal: true
+
+# A Danger script which checks out dependent projects and builds them against the
+# current (installed) version of this project.
+#
+# XXX: This script can be improved further. Open points:
+# - The script assumes that the parent project resides in the current directory
+#   and has already been installed into the local Maven repository.
+# - The script assumes this project and all downstream dependencies are Maven
+#   projects. Think of ways to generalize this.
+# - The code should be moved out of this file and molded into a proper Danger
+#   plugin.
+
+require 'nokogiri'
+require 'yaml'
+
+class Project
+  attr_reader :directory, :group_id, :artifact_id, :version
+
+  def self.load(directory)
+    File.open("#{directory}/pom.xml") do |f|
+      project = Nokogiri.XML(f).xpath('/xmlns:project')
+      parent = project.xpath('xmlns:parent')
+
+      first_present = lambda do |expr|
+        [project, parent].lazy.flat_map { |n| n.xpath(expr) }.detect(&:itself)
+      end
+
+      Project.new(
+        directory,
+        first_present.call('xmlns:groupId/text()'),
+        project.xpath('xmlns:artifactId/text()'),
+        first_present.call('xmlns:version/text()')
+      )
+    end
+  end
+
+  def initialize(directory, group_id, artifact_id, version)
+    @directory = directory
+    @group_id = group_id
+    @artifact_id = artifact_id
+    @version = version
+  end
+
+  def execute(command)
+    system("cd #{@directory} && #{command}")
```

Comment:
Wonder if you also might want to include a `pushd`/popd` or something similar here, or document the side effect. (Might be a good idea to document some of the methods, either way.)